frame/system/src/lib.rs

```diff
@@ -277,6 +265,11 @@ pub trait Trait: 'static + Eq + Clone {
 	type OnKilledAccount: OnKilledAccount<Self::AccountId>;
 
 	type SystemWeightInfo: WeightInfo;
+
+	/// Return block weights object.
+	fn block_weights() -> limits::BlockWeights {
+		Self::BlockWeights::get()
+	}
```

Comment:
it is the first time we have a function in the trait definition of a pallet I think, is it really necessary ?

Also I'm not sure it helps much compate to `T::BlockWeights::get()`

Reply:
yeah I don't really know either, anyway is fine to me